crates/goose-cli/src/cli.rs

```diff
@@ -271,6 +279,16 @@ enum Command {
         )]
         recipe: Option<String>,
 
+        #[arg(
+            long,
+            value_name = "KEY=VALUE",
+            help = "Dynamic parameters (e.g., --param username=alice)",
+            long_help = "Key-value parameters to pass to the recipe or instruction file. Can be specified multiple times.",
+            action = clap::ArgAction::Append,
+            value_parser = parse_key_val,
+        )]
+        param: Vec<(String, String)>,
```

Comment:
```suggestion
        params: Vec<(String, String)>,
```
lets make this internal variable plural to be consistent with other variables (extensions, bulitins, etc)
we can keep the flag singular